zip-0227.rst

```diff
@@ -1,8 +1,359 @@
+
+2. The `issuanceValidatingKey`, denoted as `ik`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific asset type, and is used by all blockchain users (specifically the asset owners and consensus validators) to associate the asset in question with the issuer. Throughout we also call this key the issuer verification key.
+
+The two keys are derived in an analogous manner to the `spendAuthorizingKey` and `spendValidatingKey` keys, using the same signature scheme, as described in ZIP 32 [#zip-0032]_ and in accordance with ZIP 316 [#zip-0316]_.
+
+- The `issuanceAuthorizingKey` is derived directly from the `spendingKey`, `sk`, as a private signature key:
+
+:math:`\mathsf{isk := ToScalar^{Orchard}(︀ PRF^{expand}_{sk} ([0x0a])}`
+
+- The `issuanceValidatingKey` is derived from the `issuanceAuthorizingKey`, `isk`, as a public verification key:
+
+:math:`\mathsf{ik := SpendAuthSig^{Orchard}.DerivePublic(isk)}`
+
+This allows the issuer to use the same wallet it usually uses to transfer assets, while keeping a disconnect from the other keys. Specifically, this method is aligned with the requirements and motivation of ZIP 32 [#zip-0032]_, and can further be in  It provides further anonymity and the ability to delegate issuance of an asset (or in the future, generate a multi-signature protocol) while the rest of the keys remain in the wallet safe.
+
+Asset Identifier
+----------------
+
+For every new asset, there must be a new and unique identifier of the asset. Every asset is defined by the asset description, `asset_desc`, which is a global byte string (valid across all -future- Zcash protocols). From this asset descritpion, the specific asset identifier, `AssetId`, is derived with a `GroupHash` and used within the output notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve).
+
+The main requirement for the asset identifier is for it to be collision-free, as two different issuers MUST not be able to issue the same asset type. One way to achieve this is to make the identifier dependent on the issuer keys, as the issuer keys are unique to each issuer. We achieve this by deriving the asset identifier, `AssetId` as follows:
+
+:math:`\mathsf{AssetId := GroupHash^{\mathbb{P}}}\mathsf{(ik || asset_desc)}`
+
+where
+
+- `asset_desc` is the asset description, a UTF-8 encoded string of up to 512 bytes, which includes any information pertaining to the issuance 
+- `issuanceValidatingKey` is the public key of the issuer, used to verify the signature on the transaction SIGHASH and is defined as `ik`
+
+Global Issuance State
+---------------------
+
+Issuance requires the following additions to the global state: 
+- `previously_finalized`, a set of `AssetId` that have been finalized (i.e.: the `finalize` flag has been set to `1`).
+
+Issuance Action Description
+---------------------------
```

Comment:
All serialized formats should be part of a new ZIP for the v6 transaction format. That way, we can make sure they are complete and consistent when reviewing it. (And they are easier to implement.)

You could start by copying the Orchard format from:
https://zips.z.cash/zip-0225

This can be a very rough draft, we'll fix up any issues later.

Reply:
The V6 transaction ZIP draft has been created [here](https://github.com/zcash/zips/pull/687), we will help add our additions in there via reviews/comments.